research/src/pages/select/select.proposal.mdx

```diff
@@ -201,7 +202,47 @@ and is therefore considered to be invalid.
   - Presses the escape key
   - Selects one, or more (if `multiple` is true), by clicking or hitting `enter` or `space` key
 
-### Keyboard traversal
+### Typeahead
+
+#### The <a class="link" href="#typeahead-steps" id="typeahead-steps">typeahead steps</a> given a `<select>` and a `typed character` are as follows:
+
+1. Let `start new search` be false.
+1. If longer than the <a href="#typeahead-search-timeout">typeahead search timeout</a>
+   has elapsed since the previous invocation of the [typeahead steps](#typeahead-steps)
+   for `listbox`, then set `start new search` to true and set `buffer` to an empty string.
+1. Let `buffer` be `<select>`'s [`typeahead buffer`](#typeahead-buffer)
+1. Append `typed character` to `buffer`.
+1. Let `options` be the `<select>`'s list of options.
+1. Let `focusedIndex` be the index of `<select>`'s currently focused option.
+1. If `start new search` is true, then let `options for search` be the array of options given by
+   `Array.prototype.slice.call(options, focusedIndex + 1, options.length).concat(Array.prototype.slice.call(options, 0, focusedIndex + 1))`.
+   Otherwise, let `options for search` be the array of options given by
+   `Array.prototype.slice.call(options, focusedIndex, options.length).concat(Array.prototype.slice.call(options, 0, focusedIndex))`.
+   <p class="note">
+     The point is that if we're in the middle of a search, we should keep trying to match on the
+     currently selected &lt;option&gt;. If this is a new search, we should start searching at the
+     &lt;option&gt; after the currently selected &lt;option&gt;. If the user wanted the current
+     &lt;option&gt;, then presumably they wouldn't still be typing search text.
+   </p>
+1. Let `newly focused option` be the result of [finding the first matching
```

Comment:
Do we actually want focus to move? How do we allow for keyboard entry to continue if focus has left the input and moved to the option?

Reply:
I think it should work the same way as arrow up/down, which also move focus among the options.  When thinking about the [listbox events](https://open-ui.org/components/select#part-listbox), I assume that they bubble up to the listbox when an option is focused.  Otherwise arrow up/down, Enter/Escape, etc. wouldn't work when an option is focused.

Reply:
@dandclark You're correct, I guess I'm curious about how we pass back the entered content into the text input to ensure that the experience is better. I know that currently this isn't how they work but it is what seems to be a general agreement of a better experience when offering type ahead. I think for now we can resolve on this but this will probably need to be addressed if we agree to change the anatomy to a regular input.